models/user.js

```diff
@@ -0,0 +1,46 @@
+import {
+    Schema,
+    model
+} from 'mongoose'
+import {
+    ensureFieldUniquity
+} from "./utils";
+
+import bcrypt from 'bcrypt';
+
+const UserSchema = new Schema({
+    login: {
+        type: String,
+        required: true,
+        unique: true
+    },
+    password: {
+        type: String,
+        required: true
+    },
+    role: {
+        type: Array,
+        required: true
+    },
+    isActive: {
+        type: Boolean,
+        default: false
+    }
+});
+
+const passwordHash = (ensureFieldUniquity) => {
+
+
+    if (this.password && this.isModified('password')) {
+        this.password = bcrypt.hash(this.password, 10, function (err, hash) {
```

Comment:
There is something weird in arrow functions with this - it has related to global this

https://hackernoon.com/javascript-es6-arrow-functions-and-lexical-this-f2a3e2a5e8c4